modules/custom-tuning-specification.adoc

```diff
@@ -0,0 +1,136 @@
+- match:                              # optional; if omitted, profile match is assumed unless a profile with a higher matches first
+  <match>                             # an optional array
+  priority: <priority>                # profile ordering priority, lower numbers mean higher priority (0 is the highest priority)
+  profile: <tuned_profile_name>       # e.g. tuned_profile_1
+
+# ...
+
+- match:
+  <match>
+  priority: <priority>
+  profile: <tuned_profile_name>       # e.g. tuned_profile_n
+----
+
+If `<match>` is omitted, a profile match (for example, `true`) is assumed.
+
+`<match>` is an optional array recursively defined as follows:
+
+----
+- label: <label_name>     # node or pod label name
+  value: <label_value>    # optional node or pod label value; if omitted, the presence of <label_name> is enough to match
+  type: <label_type>      # optional node or pod type ("node" or "pod"); if omitted, "node" is assumed
+  <match>                 # an optional <match> array
+----
+
+If `<match>` is not omitted, all nested `<match>` sections must also evaluate to
+`true`. Otherwise, `false` is assumed and the profile with the respective
+`<match>` section will not be applied or recommended. Therefore, the nesting
+(child `<match>` sections) works as logical and Operator. Conversely, if any
+item of the `<match>` array matches, the entire `<match>` array evaluates to
+`true`. Therefore, the array acts as logical or Operator.
+
+.Example
+
+----
+- match:
+  - label: tuned.openshift.io/elasticsearch
+    match:
+    - label: node-role.kubernetes.io/master
+    - label: node-role.kubernetes.io/infra
+    type: pod
+  priority: 10
+  profile: openshift-control-plane-es
+- match:
+  - label: node-role.kubernetes.io/master
+  - label: node-role.kubernetes.io/infra
+  priority: 20
+  profile: openshift-control-plane
+- priority: 30
+  profile: openshift-node
+----
+
+The CR above is translated for the containerized tuned daemon into its
+`recommend.conf` file based on the profile priorities. The profile with the
+highest priority (`10`) is `openshift-control-plane-es` and, therefore, it is
+considered first. The containerized tuned daemon running on a given node looks
+to see if there is a pod running on the same node with the
+`tuned.openshift.io/elasticsearch` label set. If not, the entire `<match>`
+section evaluates as false. If there is such a pod with the label, in order for
+the `<match>` section to evaluate to true, the node label also needs to be
+`node-role.kubernetes.io/master` or `node-role.kubernetes.io/infra`.
+
+If the labels for the profile with priority `10` matched,
+`openshift-control-plane-es` profile is applied and no other profile is
+considered. If the node/pod label combination did not match, the second highest
+priority profile (`openshift-control-plane`) is considered. This profile is
+applied if the containerized tuned pod runs on a node with labels
+`node-role.kubernetes.io/master` or `node-role.kubernetes.io/infra`.
+
```

Comment:
The previous paragraphs confuse me; they seem to switch back and forth between matching and not matching. Is there a flow so that _If they match, then..._ _If they don't match, then..._

Reply:
Perhaps a decision diagram would help here?  Not sure how to create one in asciidoc.  The flow is from the highest priority (from the top in the diagram below):

    priority 10
    (pod_label: tuned.openshift.io/elasticsearch) && (node_label: node-role.kubernetes.io/master || node_label: node-role.kubernetes.io/infra) --- true ---> use profile: openshift-control-plane-es
       |
     false
       |
       V  
    priority 20
    (node_label: node-role.kubernetes.io/master || node_label: node-role.kubernetes.io/infra) --- true ---> use profile: openshift-control-plane
       |
     false
       |
       V  
    priority 30 (catch-all, always true ---> use profile: openshift-node

Reply:
@jmencak Thanks! I am working with our diagrams team to see if they can create a graphic for us.

Reply:
The image is added in https://github.com/openshift/openshift-docs/pull/14056